docs(syslog): migration doc edits for syslog source (#1033)
SumoLogic/sumologic-otel-collector: docs/migration.md

    time_zone: America/Tijuana

receivers:
  ## Use a syslog receiver
  syslog:
    tcp:
      listen_address: "0.0.0.0:54526"
    protocol: rfc5424
    operators:
      - type: syslog_parser

>         protocol: rfc5424

Reviewer: We have been using tcplog, because it doesn't verify correctness of syslog format. Like Installed Collector. Installed Collector's syslog is more like tcp/udp log receiver that syslogreceiver

Reviewer: The issue here is that we want to behave as close as possible to Installed Collector, so I wouldn't try to follow the rfc in this document. We don't want to change behavior for customers.

To be honest, Syslog source in Installed Collector should be named TCP Source, but it is what it is :shrug:

Reviewer: Also we are using sumologic exporter for all of data in this document. If we use `syslog_parser` the customers will get different data that now, as now they get raw syslog lines

Author: so I tested this and it looks like the pipeline will not work without the `syslog_parser` operator at the moment

Reviewer: btw, we should fix syslog exporter to no send
`%!d(<nil>) 1970-01-01T00:00:00Z %!s(<nil>) %!s(<nil>) %!s(<nil>) %!s(<nil>) - %!s(<nil>)`

extensions:
  sumologic:
    installation_token: <installation_token>

>     api_base_url: <api_base_url>

Reviewer: We don't need it for customers. Default `api_base_url` works fine for all production deployments

  ## There is no substitute for `Description` in current project phase.
  ## It is recommended to use comments for that purpose, like this one.
  ## sumologic_syslog/<source group name>:
  ## <source group name> can be substitute of Installed Collector `Name`
  sumologic_syslog/syslog source:

>   ## re-associates record attributes to a resource to make them available to be used source attributes

Reviewer: ```suggestion
  ## re-associates record attributes to a resource to make them available to be used by Source Processor
```

  groupbyattrs:
    keys:
      - net.peer.name
      - facility

>   ## The following configuration will add two fields to every resource

Author: how do we add fields/attributes in ottl?

    add_attributes: true

processors:

>   ## The following configuration will add two fields to every resource

Reviewer: I would leave this processor, as it ensures backward compatibility with syslog source
https://github.com/SumoLogic/sumologic-otel-collector/tree/main/pkg/processor/sumologicsyslogprocessor

Please ensure that final configuration of OTC produces the same result as Installed Collector

Author: I tested this with the `syslog exporter`, I can double check it with the `sumologic exporter` again

Author: The OT collector registered successfully using this endpoint on long: https://long-open-events.sumologic.net

I saw the log message here:
https://stagdata.long.sumologic.net/ui/#/search/create?id=1ML0NOm6P7QbVhNPtWxBeunSJUhINTnfEapIjTWi

The only abnormality I noticed was that the `source name` and `source host` were `undefined`

Reviewer: We need to fix it. `facility` is probably due to missing `sumologicsyslogprocessor`

Reviewer: `net.peer.name` is probably due to `tcp`/`udp` configuration

Author: used `'groupbyattr` to fix this.

    source_name: "%{facility}"
    ## Set Source Host to `net.peer.name`
    source_host: "%{net.peer.name}"

exporters:
  sumologic/syslog:

> 

Reviewer: We should use transform processor in order to set timestamp to zero

    source_name: "%{facility}"
    ## Set Source Host to `net.peer.name`
    source_host: "%{net.peer.name}"

exporters:

>   sumologic:
>     # unique URL generated for your HTTP Source, this is the address to send data to
>     # deprecated, please use sumologicextension to manage your endpoints
>     # if sumologicextension is not being used, the endpoint is required
>     endpoint: <HTTP_Source_URL>
>     # Compression encoding format, empty string means no compression, default = gzip
>     compress_encoding: {gzip, deflate, ""}
>     # max HTTP request body size in bytes before compression (if applied),
>     # NOTE: this limit does not apply to data sent in otlp format,
>     #   to limit size of otlp requests, please use the batch processor:
>     #   https://github.com/open-telemetry/opentelemetry-collector/tree/v0.71.0/processor/batchprocessor
>     # default = 1_048_576 (1MB)
>     max_request_body_size: <max_request_body_size>
> 
>     # format to use when sending logs to Sumo Logic, default = otlp,
>     # NOTE: only `otlp` is supported when used with sumologicextension
>     log_format: {json, text, otlp}
> 

Reviewer: I would avoid adding too much configuration options for customers, epecially for migration from installed collector, as they should use defaults and shouldn't play with log format, request size, compression or endpoint by themselves

```suggestion
  sumologic:

```

  ## sumologic_syslog/<source group name>:
  ## <source group name> can be substitute of Installed Collector `Name`
  sumologic_syslog/syslog source:
  ## re-associates record attributes to a resource to make them available to be used by Source Processor
  groupbyattrs:
    keys:
      - net.peer.name
      - facility
  ## Let leave timestamp parsing to Sumo Logic backend
  transform/clear_logs_timestamp:
    log_statements:
      - context: log
        statements:
          - set(time_unix_nano, 0)
  ## The following configuration will add two fields to every resource
  transform/syslog source:
    log_statements:
      - context: resource
        statements:
          - set(attributes["cloud.availability_zone"], "zone-1")
          - set(attributes["k8s.cluster.name"], "my-cluster")
  source/syslog source:
    ## Installed Collector substitute for `Source Category`.
    source_category: example category
    ## Set Source Name to be facility name
    source_name: "%{facility}"
    ## Set Source Host to `net.peer.name`
    source_host: "%{net.peer.name}"

exporters:
  sumologic/syslog:

service:
  extensions:
  - sumologic
  pipelines:
    logs/syslog source:
      receivers:

>       - tcplog/first receiver

Reviewer: ```suggestion
      - tcplog/first receiver
      - udplog/first receiver
```

lets use both of them if we defined two

### Syslog Source (Syslog)

The equivalent of the Syslog Source is a combination of

> [the tcplog][tcplogreceiver] or [the udplog][udplogreceiver] receivers and [the sumologicsyslog processor][sumologicsyslog].

Reviewer: ```suggestion
[the Tcp][tcplogreceiver] or [the UDP][udplogreceiver] receivers and [the 
Sumo Logic Syslog Processor][sumologicsyslog].
```
